src/components/PigForm.js

```diff
@@ -0,0 +1,131 @@
+import React, { useState } from 'react'
+
+function PigForm({  setAllHogs, hogs }) {
+  const [newHogName, setNewHogName] = useState('')
+  const [newHogSpecialty, setNewHogSpecialty] = useState('')
+  const [newHogWeight, setNewHogWeight] = useState('')
+  const [newHogGreased, setNewHogGreased] = useState('')
+  const [newHogMedal, setNewHogMedal] = useState('')
+  const [newHogImg, setNewHogImg] = useState('')
+
+  function handleHogNameChange(event) {
+    setNewHogName(event.target.value)
+  }
+
+  function handleHogSpecialtyChange(event) {
+    setNewHogSpecialty(event.target.value)
+  }
+
+  function handleHogWeightChange(event) {
+    setNewHogWeight(event.target.value)
+  }
+
+  function handleHogGreasedChange(event) {
+    setNewHogGreased(event.target.value)
+  }
+
+  function handleHogsMedalChange(event) {
+    setNewHogMedal(event.target.value)
+  }
+
+  function handleHogsImgChange(event) {
+    setNewHogImg(event.target.value)
+  }
```

Comment:
If you review the form abstraction lesson you can find a much simpler way to do this.